### PR TITLE
Allow cypher result mocking for unit tests (+Move enabled check out of ArrowQueryRunner::create)

### PR DESCRIPTION
**Pull request by FlorentinD**
Thank you for your contribution to the Graph Data Science Client project.

Before submitting this PR, please read [Contributing to the Neo4j Ecosystem](https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md). 

Make sure:
- [x] You signed the [Neo4j CLA](https://neo4j.com/developer/cla/#sign-cla) (Contributor License Agreement) so that we are allowed to ship your code in our library
- [x] Your contribution is covered by tests

### CODE REVIEW

**Comment by FlorentinD** [comment]
based on feedback from #748

File: graphdatascience/query_runner/arrow_query_runner.py [hunk 1]
```diff
@@ -27,9 +27,9 @@ def create(
         disable_server_verification: bool = False,
         tls_root_certs: Optional[bytes] = None,
         connection_string_override: Optional[str] = None,
-    ) -> QueryRunner:
+    ) -> ArrowQueryRunner:
         if not arrow_info.enabled:
-            return fallback_query_runner
+            raise ValueError("Arrow is not enabled on the server")
```
Comment (Mats-SX) [inline graphdatascience/query_runner/arrow_query_runner.py:32]: I think `ValueError` is not correct to use here. I would use `Exception`.

File: graphdatascience/tests/unit/conftest.py [hunk 2]
```diff
@@ -133,12 +165,12 @@ def gds(runner: CollectingQueryRunner) -> Generator[GraphDataScience, None, None
 @pytest.fixture
 def aura_gds(runner: CollectingQueryRunner, mocker: MockerFixture) -> Generator[AuraGraphDataScience, None, None]:
     arrow_info = ArrowInfo(listenAddress="foo.bar", enabled=True, running=True, versions=[])
+    runner.add__mock_result("gds.debug.arrow", DataFrame([asdict(arrow_info)]))
```
Comment (Mats-SX) [inline graphdatascience/tests/unit/conftest.py:168]: actually much nicer than the mocker.patch IMO

File: graphdatascience/tests/unit/test_arrow_runner.py [hunk 3]
```diff
@@ -23,9 +23,8 @@ def test_create(runner: CollectingQueryRunner) -> None:
 def test_return_fallback_when_arrow_is_not_enabled(runner: CollectingQueryRunner) -> None:
     arrow_info = ArrowInfo(listenAddress="localhost:1234", enabled=False, running=False, versions=[])
 
-    arrow_runner = ArrowQueryRunner.create(runner, arrow_info)
-
-    assert arrow_runner is runner
+    with pytest.raises(ValueError, match="Arrow is not enabled"):
+        ArrowQueryRunner.create(runner, arrow_info)
```
Comment (Mats-SX) [inline graphdatascience/tests/unit/test_arrow_runner.py:27]: nice. we could remove this behaviour, but I like that you kept it.

File: graphdatascience/tests/unit/test_init.py [hunk 4]
```diff
@@ -9,7 +13,13 @@
...
 @pytest.mark.parametrize("major, minor, patch, version_string", GDS_INIT_VERSION_TESTDATA)
 def test_gds_init_version(major: int, minor: int, patch: int, version_string: str) -> None:
-    gds = GraphDataScience(CollectingQueryRunner(ServerVersion.from_string(version_string)), arrow=False)
+    arrow_info = ArrowInfo(listenAddress="foo.bar", enabled=True, running=True, versions=[])
+    query_runner = CollectingQueryRunner(
+        ServerVersion.from_string(version_string),
+        result_mock=DataFrame([asdict(arrow_info)]),
+    )
+
+    gds = GraphDataScience(endpoint=query_runner, arrow=False)
```
Comment (Mats-SX) [inline graphdatascience/tests/unit/test_init.py:22]: interesting setup. arrow is up and online, but client refuses it. it is a valid configuration.

**Review by Mats-SX — APPROVED** [review]
This was a lot! I think you made unit testing better as a result.

**Pull request merged**